utils/src/main/kotlin/Utils.kt

```diff
@@ -392,6 +392,18 @@ fun Throwable.collectMessages(): List<String> {
     return messages
 }
 
+/**
+ * Recursively collect the messages of this [Throwable] and all its causes and join them to a single [String].
+ */
+fun Throwable.collectMessagesAsString() =
```

Comment:
Couldn't we simply use `Throwable.printStackTrace(PrintWriter s)`? Also see https://stackoverflow.com/a/4812589/1127485.

Reply:
Ah, right!